System/WebPage.php

```diff
@@ -479,7 +479,7 @@ static function SkeletalShow($title, $unsupportedURL, $favIcon, $isMobileApp)
 		if(defined('FORCE_GZIP'))
 			ob_start('ob_gzhandler');
 		$symbol = empty($_GET) ? '?' : '&';
-		$url = '(document.URL.indexOf("#!/")==-1 ? document.URL.replace(location.hash||"#","")+"'.$symbol.'" : document.URL.replace("#!/","'.$symbol.'")+"&") +
+		$url = '(document.URL.indexOf("#!/")==-1 ? document.URL.replace(/#$/, "").replace(location.hash||"#","")+"'.$symbol.'" : document.URL.replace(/#$/, "").replace("#!/","'.$symbol.'")+"&") +
```

Comment:
Why isn't .replace(location.hash||"#","") enough?
So it's replacing all of the current hash, or if there is none, the "#" char, with blank
And your replace is replacing a "#", but only in the end. Seems like the previous one is even stronger.